mantidproject/mantid
# Invisible workspaces option bug fix

Description of work.

As stated in the original issue, creating hidden workspaces then checking/unchecking the "show hidden workspaces" option has no effect on those workspaces:
- If "show hidden workspaces" was already checked, unchecking it doesn't cause hidden workspaces to disappear.
- If "show hidden workspaces" was already unchecked, checking it doesn't cause hidden workspaces to appear.
- The work done has fixed this so clicking on this option produces the intended effect.

Fixes #6798.

Release notes have been updated.

**Testing Instructions**:
- Click 'Load' in the workspaces dock.
- Pick a file and make it hidden by prefixing the output workspace name with __
- There will either be a hidden workspace in the dock or no workspace depending on whether "show hidden workspaces" was already checked or not. Either way, go to View->Preferences->Mantid->Options and check/uncheck "show invisible workspaces". Then click 'Apply' or 'OK'.
- The state of the hidden workspaces should now be altered.
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
